plugins/LayerView/LayerView.qml

```diff
@@ -127,6 +127,16 @@ Item
                     } else {
                         view_settings.show_legend = false;
                     }
+
+                    typesLegenModelNoCheck.clear();
+                    typesLegenModelNoCheck.append({
+                        label: catalog.i18nc("@label", "Top / Bottom"),
+                        colorId:  "layerview_skin"
+                    });
+                    typesLegenModelNoCheck.append({
+                        label: catalog.i18nc("@label", "Inner Wall"),
+                        colorId:  "layerview_inset_x"
+                    });
```

Comment:
This `updateLegend` function is also called before the `componentCompleted`, causing these legend items to appear twice until you update for the first time.
![doublelegenditems](https://cloud.githubusercontent.com/assets/2448634/25651722/d7e68cd0-2fe4-11e7-92f8-21a6cdc1b56a.png)